core/templates/dev/head/domain/learner_dashboard/learner_dashboard_icons_directive.html

```diff
@@ -0,0 +1,23 @@
+<i class="oppia-learner-dashboard-icon fa fa-clock-o"
+   ng-show="canActivityBeAddedToLearnerPlaylist(getActivityId())"
+   ng-click="addToLearnerPlaylist(getActivityId(), getActivityType())"
+   ng-mouseenter="toggleActivityIsActive()" ng-mouseleave="toggleActivityIsActive()"
+   aria-hidden="true"
+   tooltip="<['I18N_LIBRARY_ADD_TO_LEARNER_PLAYLIST' | translate]>"
+   tooltip-placement="left"></i>
+<i class="oppia-learner-dashboard-icon fa fa-clock-o"
+   ng-if="learnerDashboardActivityIds.explorationPlaylistIds.indexOf(getActivityId()) !== -1 || learnerDashboardActivityIds.collectionPlaylistIds.indexOf(getActivityId()) !== -1"
```

Comment:
I just realized that this doesn't make sense to me. Don't you already know the activity type? In that case you should be doing just the appropriate check. Explorations and collections could (unlikely, but it's possible) have the same ID.

I suggest you pull this check out into a standalone function within the directive controller and do it properly based on the activity type. Ditto below.

Reply:
Done!